client/utils.c

```diff
@@ -364,53 +350,55 @@ TDNFIsDir(
     goto cleanup;
 }
 
-//update time if file exists
-//create if not
+/* update time if file exists, create if not */
 uint32_t
 TDNFTouchFile(
     const char* pszFile
     )
 {
+    int fd = 0;
+    mode_t old_mask;
     uint32_t dwError = 0;
-    struct stat st = {0};
-    int fd = -1;
-    struct utimbuf times = {0};
 
-    if(IsNullOrEmptyString(pszFile))
+    if (IsNullOrEmptyString(pszFile))
     {
         dwError = ERROR_TDNF_INVALID_PARAMETER;
-        BAIL_ON_TDNF_ERROR(dwError);
+        goto cleanup;
     }
 
+    old_mask = umask(022);
     fd = creat(pszFile, S_IRUSR | S_IRGRP | S_IROTH);
-    if(fd == -1)
+    if (fd < 0)
     {
-        if(errno == EEXIST)
+        if (errno == EEXIST)
         {
+            struct stat st = {0};
+            struct utimbuf times = {0};
+
             times.actime = st.st_atime;
             times.modtime = time(NULL);
-            if(utime(pszFile, &times))
+            if (utime(pszFile, &times))
             {
                 dwError = errno;
-                BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+                goto cleanup;
```

Comment:
This would make the function return an errno as a TDNF error. The `BAIL_ON_TDNF_SYSTEM_ERROR` macro adds a base value to it.

But I also don't like this also for other reasons. The `BAIL_ON_TDNF_ERROR` has the advantage that it can be redefined to do more work. For example, sometimes I change it to print out a message, which makes it easier to find the trigger for an error code.